pkg/cfn/manager/nodegroup.go

```diff
@@ -99,7 +105,39 @@ func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(ctx context.Con
 			asgNames = append(asgNames, *asg.Name)
 		}
 	}
-	return c.PropagateManagedNodeGroupTagsToASG(ng.Name, ng.Tags, asgNames, errorCh)
+
+	// add labels and taints
+	tags, err := convertLabelsAndTaintsIntoTags(ng)
+	if err != nil {
+		return err
+	}
+
+	// add nodegroup tags
+	for k, v := range ng.Tags {
+		tags[k] = v
+	}
+
+	return propagateFunc(ng.Name, tags, asgNames, errorCh)
+}
+
+// convertLabelsAndTaintsIntoTags so that they can be propageted automatically to ASG afterwards
+func convertLabelsAndTaintsIntoTags(ng *api.ManagedNodeGroup) (map[string]string, error) {
+	result := make(map[string]string, 0)
+
+	// labels
+	for k, v := range ng.Labels {
+		result[labelsPrefix+k] = v
+	}
+
+	// taints
+	for _, taint := range ng.Taints {
+		if _, ok := ng.Labels[taint.Key]; ok {
+			return nil, fmt.Errorf("duplicate key found for taints and labels with taint key=value: %s=%s, and label: %s=%s", taint.Key, taint.Value, taint.Key, ng.Labels[taint.Key])
```

Comment:
it'll be good to add a test for this error as well, wdyt? 💡

Reply:
74cde341357fe3e2a32f2ebc5e757c45f37d128e